Issue 130: Transaction Propagation

mmusgrov opened the pull request:
Signed-off-by: Michael Musgrove <mmusgrov@redhat.com>
#130 
Include tests that validate that a ManagedExecutor is able to propagate transaction context to dependent stages. The test also tests TransactionScoped beans

mmusgrov commented:
I have just squashed my commits for this PR

njr-11 commented on tck/src/main/java/org/eclipse/microprofile/concurrency/tck/cdi/JTACDITest.java:
```diff
@@ -0,0 +1,310 @@
+        }
+    }
+
+    public void testRunWithTxnOfExecutingThread() throws SystemException, NotSupportedException {
```
This test is missing the `@Test` annotation so it doesn't run.  If it did run, it would fail.  I think this test has a bug in that, if it aims to run under the transaction that is activate on the thread at the time when the task runs (as the method name suggests), the configuration should have `.propagated().unchanged(ThreadContext.TRANSACTION).cleared(ThreadContext.ALL_REMAINING)`

mmusgrov replied:
Oops, and thanks for spotting the bug too.

mmusgrov commented on tck/src/main/java/org/eclipse/microprofile/concurrency/tck/cdi/JTACDITest.java:
```diff
@@ -0,0 +1,310 @@
+    @Deployment
+    public static WebArchive createDeployment() {
+        // build a JAR that provides three fake context types: 'Buffer', 'Label', and 'ThreadPriority'
+        JavaArchive fakeContextProviders = ShrinkWrap.create(JavaArchive.class, "fakeContextTypes.jar")
```
+1

mmusgrov commented on tck/src/main/java/org/eclipse/microprofile/concurrency/tck/cdi/JTACDITest.java:
```diff
@@ -0,0 +1,310 @@
+    private TransactionalService transactionalService;
+
+    @Inject
+    private TransactionManager transactionManager;
```
This makes sense and thanks for the alternative.

mmusgrov commented:
@njr-11 I have updated these tests for the optional feature of transaction propagation to use UserTransaction. I also updated the example in the documentation PR https://github.com/eclipse/microprofile-concurrency/pull/132 to also use UserTransaction.

mmusgrov commented:
Okay thanks. I have just squashed the commits relating to this PR since it seems ready.
